src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1187 @@
 <?php
 
-/*
+/**
  * This file is part of the Behat\Mink.
  * (c) Konstantin Kudryashov <ever.zet@gmail.com>
  *
  * For the full copyright and license information, please view the LICENSE
  * file that was distributed with this source code.
+ *
+ * @noinspection PhpLanguageLevelInspection
  */
 
 namespace Mink\WebdriverClassDriver;
 
 use Behat\Mink\Driver\CoreDriver;
+use Behat\Mink\Exception\DriverException;
+use Facebook\WebDriver\Exception\NoSuchCookieException;
+use Facebook\WebDriver\Exception\NoSuchElementException;
+use Facebook\WebDriver\Exception\WebDriverException;
+use Facebook\WebDriver\Remote\DesiredCapabilities;
+use Facebook\WebDriver\Remote\RemoteWebDriver;
+use Facebook\WebDriver\Remote\RemoteWebElement;
+use Facebook\WebDriver\Remote\WebDriverBrowserType;
+use Facebook\WebDriver\WebDriverBy;
+use Facebook\WebDriver\WebDriverDimension;
+use Facebook\WebDriver\WebDriverElement;
+use Facebook\WebDriver\WebDriverRadios;
+use Facebook\WebDriver\WebDriverSelect;
+use JetBrains\PhpStorm\Language;
+use JsonException;
+use Throwable;
 
 class WebdriverClassicDriver extends CoreDriver
 {
+    public const DEFAULT_BROWSER = 'chrome';
+
+    public const DEFAULT_CAPABILITIES = [
+        'default' => [
+            'platform' => 'ANY',
+            'name' => 'Behat Test',
+            'deviceOrientation' => 'landscape',
+            'deviceType' => 'desktop',
+        ],
+
+        'chrome' => [
+            'goog:chromeOptions' => [
+                // disable "Chrome is being controlled.." notification bar
+                'excludeSwitches' => ['enable-automation'],
+            ],
+        ],
+
+        'firefox' => [
+        ],
+
+        'edge' => [
+            'ms:edgeOptions' => [
+                // disable "Microsoft Edge is being controlled.." notification bar
+                'excludeSwitches' => ['enable-automation'],
+                // disable menu shown when text is selected (which interferes with double-clicking)
+                'prefs' => [
+                    'edge_quick_search.show_mini_menu' => false,
+                ],
+            ],
+        ],
+    ];
+
+    private const W3C_WINDOW_HANDLE_PREFIX = 'w3cwh:';
+
+    private ?RemoteWebDriver $webDriver = null;
+
+    private string $browserName;
+
+    private DesiredCapabilities $desiredCapabilities;
+
+    private array $timeouts = [];
+
+    private string $webDriverHost;
+
+    private ?string $initialWindowName = null;
+
+    /**
+     * @throws DriverException
+     */
+    public function __construct(
+        string $browserName = self::DEFAULT_BROWSER,
+        array $desiredCapabilities = [],
```

Comment:
should we support passing a DesiredCapabilities object directly ?

Reply:
Probably not:
- that class is not part of the public interface of this library
  - the right way would be to provide our own (perhaps a subclass of the original)
- we wouldn't be able to add in the default browser-specific capabilities

Reply:
Also, partly related: apparently the concept of "desired" and "required" capabilities is now deprecated, so we should also avoid that naming within the driver and especially the public interface (I changed this already).